cloudsync_onedrive.py

```diff
@@ -326,46 +209,44 @@ def __init__(self, oauth_config: Optional[OAuthConfig] = None):
         self.__drive_by_id: Dict[str, Drive] = {}
         self.__site_by_id: Dict[str, Site] = {}
         self.__cached_is_biz = None
-        self._http = HttpProvider()
-
+        self._http = requests.Session()
 
     @property
     def connected(self):
         return self.__client is not None
 
-    @staticmethod
-    def _ensure_event_loop():
-        try:
-            asyncio.get_event_loop()
-        except RuntimeError:
-            asyncio.set_event_loop(asyncio.new_event_loop())
+    def _api_path(self, *, oid: Optional[str]=None, path: Optional[str]=None) -> str:
+        assert oid or path
+
+        if oid == "root" or path == "/":
+            oid = self.namespace.api_root_oid
+
+        if oid:
+            return f"/drives/{self._validated_namespace_id}/items/{oid}"
 
-    def _get_url(self, api_path):
-        api_path = api_path.lstrip("/")
-        with self._api() as client:
-            return client.base_url.rstrip("/") + "/" + api_path
+        enc_path = urllib.parse.quote(path)
+        return f"/drives/{self._validated_namespace_id}/{self.namespace.api_root_path}:{enc_path}:"
 
     # names of args are compat with requests module
     def _direct_api(self, action, path=None, *, url=None, stream=None, data=None, headers=None,
             json=None, raw_response=False, timeout=SOCK_TIMEOUT):  # pylint: disable=redefined-outer-name
         assert path or url
 
         if not url:
-            url = self._get_url(path)
+            url = self._base_url.rstrip("/") + "/" + path.lstrip("/")
```

Comment:
isn't there already some sort of join method or function that joins paths this way?

Reply:
yeah, but it is meant for file paths, don't think we should be using it for urls.